### ISSUE TITLE
Troubles with having multiple wildcard dimension names

### ISSUE THREAD

**Issue by oxinabox** [invenia/NamedDims.jl#8]
This issue will be/was introduced with #7 
I think this is an acceptable issue for now really.
As once you have multiple wildcards then you need to be in the case of a 3+ dimension array to have any nonwild card dimensions left that you would want to refer to by name.
Which is fairly rare.

MWE for no duplicate wildcards.
It mostly works fine,
until you try and access any of the named dimensions.

```
julia> nda = NamedDimsArray{(:_,:_,:c)}(ones(10,20,30));

julia> nda[c=2]
ERROR: duplicate field name in NamedTuple: "_" is not unique
Stacktrace:
 [1] default_inds(::Tuple{Symbol,Symbol,Symbol}) at /Users/oxinabox/.julia/dev/NamedDims/src/name_core.jl:61
 [2] #order_named_inds#5(::Base.Iterators.Pairs{Symbol,Int64,Tuple{Symbol},NamedTuple{(:c,),Tuple{Int64}}}, ::Function, ::Tuple{Symbol,Symbol,Symbol}) at /Users/oxinabox/.julia/dev/NamedDims/src/name_core.jl:82
 [3] #order_named_inds at ./none:0 [inlined]
 [4] #order_named_inds#19 at /Users/oxinabox/.julia/dev/NamedDims/src/wrapper_array.jl:134 [inlined]
 [5] #order_named_inds at ./none:0 [inlined]
 [6] #getindex#20 at /Users/oxinabox/.julia/dev/NamedDims/src/wrapper_array.jl:143 [inlined]
 [7] (::getfield(Base, Symbol("#kw##getindex")))(::NamedTuple{(:c,),Tuple{Int64}}, ::typeof(getindex), ::NamedDimsArray{(:_, :_, :c),Float64,3,Array{Float64,3}}) at ./none:0
 [8] top-level scope at none:0

julia> sum(nda; dims=:c)
ERROR: duplicate field name in NamedTuple: "_" is not unique
```

**Comment by oxinabox** [comment]
Originally posted by @mcabbott  in #4
but will be solved by the same solution as to fix the above problem:
 
> I was about to open an issue, but perhaps this isn't expected to work yet... anyway this broadcasting operation gives an error:
> ```
> julia> nda = NamedDimsArray{(:x,:y,:z)}(rand(10,20,30)); # as in readme, but Array not Arrays
> 
> julia> nda .+ rand(20)'
> ERROR: duplicate field name in NamedTuple: "_" is not unique
> Stacktrace:
>  [1] identity_namedtuple(::Tuple{Symbol,Symbol}) at /Users/me/.julia/packages/NamedDims/JWdM2/src/name_core.jl:78
>  [2] combine_names_longest(::Tuple{Symbol,Symbol,Symbol}, ::Tuple{Symbol,Symbol}) at /Users/me/.julia/packages/NamedDims/JWdM2/src/name_core.jl:201
>  [3] broadcasted_names at /Users/me/.julia/packages/NamedDims/JWdM2/src/broadcasting.jl:20 [inlined]
>  ```
> Curiously it doesn't happen for 3 dimensions:
> ```
> julia> nda .+ rand(1,1,30);
> 
> julia> nda .+ rand(10);
> 
> julia> nda .+ rand(1,20);
> ERROR: duplicate field name in NamedTuple: "_" is not unique
> ```


The fact that this can occur in 2D for addition actually makes this more significant so should work on this sooner ratehr than later

**Comment by nickrobinson251** [comment]
Was this closed by #21 ? :)

**Comment by oxinabox** [comment]
Yes
